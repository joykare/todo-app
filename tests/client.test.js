import React from "react";
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from "../client/components/app.js";

describe("Tests for <App /> Component", () => {
  it("takes a snapshot of the app component", () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("initially has empty state", () => {
    const wrapper = shallow(<App />);

    expect(wrapper.state().todos).toEqual([]);
    expect(wrapper.state().title).toEqual("");
    expect(wrapper.state().description).toEqual("");
    expect(wrapper.state().date).toEqual("");
    expect(wrapper.find('.loading').length).toEqual(1);
  })

  it("when state is set it displays list of todos", () => {
    const wrapper = shallow(<App />);

    wrapper.setState({
      todos: [{
        title: "testy westy",
        description: "test 1 going on",
        date: new Date("1.01.1991")
      }]
     });

    expect(wrapper.find('.list-group').length).toBeGreaterThanOrEqual(1);
    expect(wrapper.find('.loading').length).toEqual(0);
  })
})