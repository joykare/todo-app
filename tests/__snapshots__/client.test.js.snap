// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <App /> Component takes a snapshot of the app component 1`] = `
<div>
  <div
    className="text-center"
  >
    <h4>
      Todo App
    </h4>
  </div>
  <hr />
  <div
    className="jumbotron"
    style={
      Object {
        "marginLeft": 250,
        "marginRight": 250,
      }
    }
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label>
          Title:
        </label>
        <input
          className="form-control"
          onBlur={[Function]}
          placeholder="Enter todo title"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Description
        </label>
        <input
          className="form-control"
          onBlur={[Function]}
          placeholder="Enter todo description"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Due Date:
        </label>
        <input
          className="form-control"
          onBlur={[Function]}
          placeholder="Enter date in the format mm.dd.yyyy"
          type="text"
        />
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Add Todo
      </button>
    </form>
  </div>
  <div
    className="loading"
  >
    Loading..
  </div>
</div>
`;
